lib/capistrano/tasks/deploy.rake

```diff
@@ -253,6 +255,24 @@ namespace :deploy do
     end
   end
 
+  desc "Place a REVISION_TIME file with the current revision commit time in the current release path"
+  task :set_current_revision_time do
+    on release_roles(:all) do
+      within release_path do
+        execute :echo, "\"#{fetch(:current_revision_time)}\" > REVISION_TIME"
```

Comment:
❓ Should this be skipped if `fetch(:current_revision_time)` is nil?

Reply:
I don't mind making this conditional, but just skipping entirely would mean that if the revision time couldn't be gotten for some reason in this deployment, the previous revision time would be retained whereas piping an empty string to the file will clear it:

```
~ via 🐍 v3.10.12 via 💎
❯ echo 'hello world' > file

~ via 🐍 v3.10.12 via 💎
❯ cat file
hello world

~ via 🐍 v3.10.12 via 💎
❯ echo '' > file

~ via 🐍 v3.10.12 via 💎
❯ cat file

```

While in theory it should always work or not work, I think it's nice if we have some just-in-case logic for this given it should be cheap - with that in mind do you have a preference for:
  - always outputting to the `REVISION_TIME` file, or
  - always remove the `REVISION_TIME` file first then echo conditionally, or
  - remove the `REVISION_TIME` file if it exists then do the echo conditionally

Reply:
I'm not sure I understand this part:

> if the revision time couldn't be gotten for some reason in this deployment, the previous revision time would be retained

When `cap deploy` runs, it creates a new release directory, so if nothing else happens, then there will be no `REVISION_TIME` present. In what circumstances would the previous release's `REVISION_TIME` file end up in the newly created release?

Regardless, I think what I'm looking for is for the `REVISION_TIME` file to be absent if something goes wrong, or if a non-supported SCM is being used. I'm worried about Capistrano adding empty files into users' releases that serve no apparent purpose.

Reply:
> When cap deploy runs, it creates a new release directory

Yeah sorry I swear it's being a day and a half today... I was thinking of the `revisions.log` which is kept in the root directory whereas `REVISION` (and now `REVISION_TIME`) are kept in the directory for each release, which makes tons of sense because of this very situation